### PR TITLE
[Improve][transform-v2] Support dynamic types for array function

### PR DESCRIPTION
**Pull request by CosmosNi**
Array function support string, double, long, int types.
Currently, sql parse does not support float and boolean.

### Check list

* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update [plugin-mapping.properties](https://github.com/apache/seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  2. Update the pom file of [seatunnel-dist](https://github.com/apache/seatunnel/blob/dev/seatunnel-dist/pom.xml)
  3. Add ci label in [label-scope-conf](https://github.com/apache/seatunnel/blob/dev/.github/workflows/labeler/label-scope-conf.yml)
  4. Add e2e testcase in [seatunnel-e2e](https://github.com/apache/seatunnel/tree/dev/seatunnel-e2e/seatunnel-connector-v2-e2e/)
  5. Update connector [plugin_config](https://github.com/apache/seatunnel/blob/dev/config/plugin_config)
* [ ] Update the [`release-note`](https://github.com/apache/seatunnel/blob/dev/release-note.md).

### CODE REVIEW

File: seatunnel-e2e/seatunnel-transforms-v2-e2e/seatunnel-transforms-v2-e2e-part-2/src/test/resources/sql_transform/func_array.conf [hunk 1]
```diff
@@ -25,11 +25,10 @@ env {
 
 source {
   FakeSource {
-      plugin_output = "fake"
+    result_table_name = "fake"
```
Comment (Hisoka-X) [inline seatunnel-e2e/seatunnel-transforms-v2-e2e/seatunnel-transforms-v2-e2e-part-2/src/test/resources/sql_transform/func_array.conf]: we use `plugin_output` at now.

**Review by Hisoka-X — COMMENTED** [review]
I find current code doesn't support create an array with table field, for example:

The fake table contains:
| id | name | age |
| --- | --- | --- |
| 1 | tom | 20 |
| 2 | jack | 21 |

The sql is
```
SELECT Array('c_1', age) as string_array from fake,
```

The result should be:

| string_array |
| --- |
| ['c_1', '20'] |
| ['c_1', '21'] |

**Comment by CosmosNi** [comment]
@Hisoka-X I've also considered.I'm still looking for how  to get field value

**Comment by CosmosNi** [comment]
@Hisoka-X PTAL

**Comment by Hisoka-X** [comment]
Please resolve conflict. Thanks

**Review by Hisoka-X — APPROVED** [review]
LGTM. Thanks @CosmosNi ! cc @rewerma

**Review by hailin0 — APPROVED** [review]

**Pull request merged**